elyra/pipeline/catalog_connector.py

```diff
@@ -41,23 +39,9 @@ class ComponentCatalogConnector(LoggingConfigurable):
     Abstract class to model component_entry readers that can read components from different locations
     """
 
-    # Data structure to encapsulate various capabilities & flags for a connector class
-    configuration = DictTrait(
-        default_value={
-            "max_readers": 3  # Sets the number of reader threads in read_component_definitions()
-        },
-        per_key_traits={
-            "max_readers": Integer()
-        },
-        key_trait=CUnicode(),
-        help="""A dictionary of configurable settings for a ComponentCatalogConnector class.
-        Subclasses can override these settings as needed.
-
-        Settings:
-            'max_readers': Integer; sets the maximum number of reader threads to be used to read
-                           catalog entries in parallel in read_component_definitions(); default 3
-        """
-    ).tag(config=True)
+    max_readers = Integer(3, config=True, allow_none=True,
```

Comment:
This looks good.  Since we see more containerized installations, I think it would be good to generally provide env variable support for configurable values.  This can be in a follow-up, but I think we'd add an `ELYRA_MAX_CATALOG_CONNECTOR_READERS` (or something named similarly) that provides the "default value" (and this env defaults to 3 if not present).  An example of this can be found where we deal with the [cache ttl value](https://github.com/elyra-ai/elyra/blob/28c3eab3fe55064dd5ee8f3be24e733f00d37519/elyra/pipeline/component_registry.py#L78-L91).

Reply:
Sounds good! I can open an issue for that